worker/dockerworker/dockerworker.go

```diff
@@ -96,12 +117,18 @@ func (d *dockerworker) StartWorker(state *run.State) (protocol.Transport, error)
 
 	transp := protocol.NewStdioTransport()
 
+	workerLocation, err := getWorkerLocation(state.CloudProviderName, state.WorkerConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Could not encode the worker location: %v", err)
+	}
+
 	// the --host taskcluster-worker-runner instructs docker-worker to merge
 	// config from $DOCKER_WORKER_CONFIG.
 	mainJs := fmt.Sprintf("%s/src/bin/worker.js", d.wicfg.Path)
 	cmd := exec.Command("node", mainJs, "--host", "taskcluster-worker-runner", "production")
 	cmd.Env = os.Environ()
 	cmd.Env = append(cmd.Env, "DOCKER_WORKER_CONFIG="+d.wicfg.ConfigPath)
+	cmd.Env = append(cmd.Env, "TASKCLUSTER_WORKER_LOCATION="+workerLocation)
```

Comment:
Rather than pass this one piece of configuration as an env var, let's just put this in the config file like the rest of the config.  So in `ConfigureRun` there'd be `set("workerLocation", workerLocationJson)`.  Then docker-worker would create the necessary environment variable with that value.